src/overlays/actors/ovl_En_Door_Etc/z_en_door_etc.c

```diff
@@ -27,42 +32,223 @@ const ActorInit En_Door_Etc_InitVars = {
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_U8(targetMode, 0, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_U16(shape.rot.x, 0, ICHAIN_CONTINUE),
     ICHAIN_U16(shape.rot.z, 0, ICHAIN_STOP),
 };
 
-#endif
-
-extern ColliderCylinderInit D_80AC25A0;
-extern InitChainEntry D_80AC2634[];
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/EnDoorEtc_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/EnDoorEtc_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC2044.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC20A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC2118.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC2154.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC21A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC2354.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/EnDoorEtc_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Door_Etc/func_80AC24A8.s")
+void EnDoorEtc_Init(Actor* thisx, PlayState* play2) {
+    PlayState* play = play2;
+    s32 objectIndex;
+    EnDoorEtcInfo* objectInfo = sObjInfo;
+    s32 i;
+    EnDoorEtc* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actor.shape.rot.x = -0x4000;
+    this->angle = 0;
+    for (i = 0; i < 0xF; i++, objectInfo++) {
+        if (play->sceneNum == objectInfo->sceneNum) {
+            break;
+        }
+    }
+    if ((i >= 0xF) && (Object_GetIndex(&play->objectCtx, 2) >= 0)) {
+        objectInfo++;
+    }
+    objectIndex = Object_GetIndex(&play->objectCtx, objectInfo->objectId);
+    if (objectIndex < 0) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        this->objectIndex = objectIndex;
+        this->dListIndex = objectInfo->dListIndex;
+        if (this->actor.objBankIndex == this->objectIndex) {
+            func_80AC20A8(this, play);
+        } else {
+            this->actionFunc = func_80AC20A8;
+        }
+    }
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    this->actor.colChkInfo.mass = 0xFF;
+}
+
+void EnDoorEtc_Destroy(Actor* thisx, PlayState* play) {
+    EnDoorEtc* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+/**
+ * Calculates if the distance between `a` and `b` are greater than distance `c`
+ */
+s32 EnDoorEtc_IsDistanceGreater(Vec3f* a, Vec3f* b, f32 c) {
+    f32 dx = b->x - a->x;
+    f32 dy = b->y - a->y;
+    f32 dz = b->z - a->z;
+
+    return ((SQ(dx) + SQ(dy) + SQ(dz)) < (SQ(c)));
+}
+
+void func_80AC20A8(EnDoorEtc* this, PlayState* play) {
+    if (Object_IsLoaded(&play->objectCtx, this->objectIndex)) {
+        this->actor.flags &= ~FLAGS;
```

Comment:
I think it would be better to use `ACTOR_FLAG_10` instead of all of this actor's flags